test/integrations/database/MemoryDb.spec.ts

```diff
@@ -0,0 +1,402 @@
+
+            expect(result.length).toBe(2);
+            expect(result[0]).toEqual(expect.objectContaining(margherita));
+            expect(result[1]).toEqual(expect.objectContaining(pepperoni));
+        });
+
+        it('should find no records that match the AND condition', async () =>
+        {
+            const query: RecordQuery = 
+            {
+                'AND':
+                [
+                    {
+                        'name': { EQUALS: 'Hawai' },
+                        'size': { NOT_EQUALS: 20 }
+                    }
+                ]
+            };
+
+            const result = await database.searchRecords(RECORD_TYPE_PIZZA, query);
+
+            expect(result[0]).toBe(undefined);
+        });
+
+        it('should find no records that match the OR condition', async () =>
+        {
+            const query: RecordQuery = 
+            {
+                'OR':
+                [
+                    { 'name': { EQUALS: 'Romanos' } }, 
+                    { 'size': { EQUALS: 21 } }
+                ]
+            };
+
+            const result = await database.searchRecords(RECORD_TYPE_PIZZA, query);
+ 
+            expect(result[0]).toBe(undefined);
+        });
+    });
+
+    describe('.sortRecords', () =>
+    {
+        it('should sort records asending', async () =>
+        {
+            const query: RecordQuery = { };
+            const sort: RecordSort = { 'name': SortDirections.ASCENDING };
+            const result = await database.searchRecords(RECORD_TYPE_PIZZA, query, undefined, sort);
+
+            expect(result.length).toBe(5);
+            expect(result[0]).toEqual(expect.objectContaining(calzone));
+            expect(result[1]).toEqual(expect.objectContaining(hawai));
+            expect(result[2]).toEqual(expect.objectContaining(margherita));
+            expect(result[3]).toEqual(expect.objectContaining(pepperoni));
+            expect(result[4]).toEqual(expect.objectContaining(vegetario));
+        });
+
+        it('should sort the records descending', async () =>
+        {
+            const query: RecordQuery = { };
+            const sort: RecordSort = { 'size': SortDirections.DESCENDING };
+            const result = await database.searchRecords(RECORD_TYPE_PIZZA, query, undefined, sort);
+            
+            expect(result.length).toBe(5);
+            expect(result[0]).toEqual(expect.objectContaining(vegetario));
+            expect(result[1]).toEqual(expect.objectContaining(calzone));
+            expect(result[2]).toEqual(expect.objectContaining(hawai));
+            expect(result[3]).toEqual(expect.objectContaining(pepperoni));
+            expect(result[4]).toEqual(expect.objectContaining(margherita));
+        });
+
+        it('should sort the records by multiple fields same direction', async () =>
+        {
+            const query: RecordQuery = { };
+            const sort: RecordSort = { 'size': SortDirections.DESCENDING, 'name': SortDirections.DESCENDING };
```

Comment:
The result of this sort is non transparent. I expect this to be in visible in the fixture.

Reply:
All mutiple results specified in fixture.